target_stitch/__init__.py

```diff
@@ -599,24 +607,26 @@ def handle_line(self, line):
         if isinstance(message, singer.SchemaMessage):
             self.flush()
 
+            if message.stream not in self.messages:
+                self.messages[message.stream] = []
             self.stream_meta[message.stream] = StreamMeta(
                 message.schema,
                 message.key_properties,
                 message.bookmark_properties)
 
         elif isinstance(message, (singer.RecordMessage, singer.ActivateVersionMessage)):
-            if self.messages and (
-                    message.stream != self.messages[0].stream or
-                    message.version != self.messages[0].version):
+            current_stream = message.stream
+            if self.messages[current_stream] and (
+                    message.version != self.messages[current_stream][0].version):
                 self.flush()
```

Comment:
This used to be "flush when the stream changes or when the version changes, but is now just when the version changes. Could we document this behavior some with a comment.

Reply:
Added a comment above the relevant code. Do you want this documented somewhere else too?